frame/nomination-pools/src/lib.rs

```diff
@@ -3169,12 +3191,17 @@ impl<T: Config> Pallet<T> {
 				"depositor must always have MinCreateBond stake in the pool, except for when the \
 				pool is being destroyed and the depositor is the last member",
 			);
+			expected_tvl += T::Staking::total_stake(&bonded_pool.bonded_account())
+				.expect("all pools must have total stake");
```

Comment:
Rather than spread the try_state logic for expected TVL across multiple areas of the try-state hook, could you please move it all together? 

It's fine to iterate over pools again for this purpose.

Reply:
I put the TVL checks into one place now but the check for TVL integrity is now done via the PoolMembers instead of all BondingPools. I would love some feedback on the way it's done now.